pages/Design-Patterns.md

```diff
@@ -51,58 +52,58 @@ $veyron = AutomobileFactory::create('Bugatti', 'Veyron');
 print_r($veyron->getMakeAndModel()); // outputs "Bugatti Veyron"
 {% endhighlight %}
 
-This code uses a factory to create the Automobile object. There are two possible benefits to building your code this
-way; the first is that if you need to change, rename, or replace the Automobile class later on you can do so and you
-will only have to modify the code in the factory, instead of every place in your project that uses the Automobile class.
-The second possible benefit is that if creating the object is a complicated job you can do all of the work in the
-factory, instead of repeating it every time you want to create a new instance.
+এই কোডটি অটোমোবাইল অবজেক্ট  তৈরির জন্য একটি ফ্যাক্টরি ব্যবহার করে। এইভাবে আপনার কোডটি তৈরীর জন্য এই দুটি সম্ভাব্য সুবিধা
+রয়েছে; প্রথমটি হলো পরবর্তীতে যদি কখনো আপনি অটোমোবাইল ক্লাসটি পরিবর্তন, নাম পরিবর্তন অথবা রিপ্লেস করতে চান তাহলে শুধু ফ্যাক্টরি
+ক্লাসে পরিবর্তন করলেই চলবে, সমস্ত প্রজেক্টের যত জায়গায় অটোমোবাইল ক্লাস ব্যবহার করা হয়েছে সব জায়গায় পরিবর্তনের পরিবর্তে।
+দ্বিতীয় সম্ভাব্য সুবিধা হল যে যদি অবজেক্ট তৈরি করা একটি জটিল কাজ হয় তাহলে আপনি সব কাজ করতে পারেন এই ফ্যাক্টরি ক্লাসে, প্রতিবার 
+অবজেক্ট তৈরী করার সময়ে করার পরিবর্তে।  
 
-Using the factory pattern isn't always necessary (or wise). The example code used here is so simple that a factory
-would simply be adding unneeded complexity. However if you are making a fairly large or complex project you may save
-yourself a lot of trouble down the road by using factories.
+ফ্যাক্টরি প্যাটার্ন ব্যবহার করা সবসময় (বা বুদ্ধিমান) প্রয়োজন নাও হতে পারে। এখানে ব্যবহৃত উদাহরণ কোড এত সহজ যে ফ্যাক্টরি ক্লাস এখানে 
+সহজেই অপ্রয়োজনীও জটিলতা যোগ করবে। যাইহোক, যদি আপনি একটি বেশ বড় বা জটিল প্রজেক্টে কাজ করেন তাহলে ফ্যাক্টরি ক্লাসের ব্যবহার
+হয়তো আপনার ভবিষৎ অনেক সমস্যা কমিয়ে দিবে।
 
-* [Factory pattern on Wikipedia](https://en.wikipedia.org/wiki/Factory_pattern)
 
-## Singleton
+* [ফ্যাক্টরি প্যাটার্ন উইকিপিডিয়াতে](https://en.wikipedia.org/wiki/Factory_pattern)
 
-When designing web applications, it often makes sense conceptually and architecturally to allow access to one and only
-one instance of a particular class. The singleton pattern enables us to do this.
+## সিঙ্গেলটন প্যাটার্ন
 
-**TODO: NEED NEW SINGLETON CODE EXAMPLE**
+যখন ওয়েব অ্যাপ্লিকেশন ডিজাইন করা হয় তখন প্রায়ই ধারনাগত এবং স্থাপত্যগতভাবে একটি নির্দিষ্ট ক্লাসের এক এবং শুধুমাত্র একটি ইন্সটান্স আক্সেস
+করার প্রয়োজন হতে পারে। সিঙ্গেলটন প্যাটার্ন আমাদের এটা করতে সহায়তা করে।
 
-The code above implements the singleton pattern using a [*static* variable](http://php.net/language.variables.scope#language.variables.scope.static) and the static creation method `getInstance()`.
-Note the following:
+**যা করতে হবে : নতুন সিঙ্গেলটন কোড উদাহরণ প্রয়োজন **
 
-* The constructor [`__construct()`](http://php.net/language.oop5.decon#object.construct) is declared as protected to
```

Comment:
@ajaxray vai, can you please check this sentence??
It seems the `__construct()` function should declare as a `private` rather than `protected`

Reply:
Actually,  it seems to be right. Because by using `protected` instead of `private`, it's fulfilling the purpose of **prevent creating a new instance outside of the class** while  **allows the subclassing of the class `Singleton`**

Reply:
I found few others places where they suggest to make the `__constructor()` private, that's why I confused.